### PR TITLE
Check if a number is Odd or Even 

### PR DESCRIPTION
**Pull request by Bikaze**
Implemented a solution for #60

### CODE REVIEW

File: solutions/check_odd_or_even.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python3
+"""
+A module for determining if a number is even or odd.
```
Comment (ObayCipher) [inline solutions/check_odd_or_even.py:3]: add this:
Author: 
Date: 
Group:

Comment (Bikaze) [reply solutions/check_odd_or_even.py:3]: Done!

File: solutions/check_odd_or_even.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+Module contents:
+    - check_odd_or_even: Returns "Even" if the number is even, "Odd" otherwise.
+"""
+
+
+def check_odd_or_even(num):
```
Comment (ObayCipher) [inline solutions/check_odd_or_even.py:14]: Missing type annotations for the function:
```python

def check_odd_or_even(num: int) -> str:

```

File: solutions/check_odd_or_even.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    Checks if the provided integer is even or odd.
+
+    Parameters:
+        num (int): The integer to check.
+
+    Returns:
+        str: "Even" if num is even, otherwise "Odd".
+
+    Raises:
+        AssertionError: If num is not an integer.
```
Comment (ObayCipher) [inline solutions/check_odd_or_even.py:25]: Exception part:
The function docstring mentions the `AssertionError`, **but** the description in the **Raises:** section could be clearer. It's  **important** to state that an ```AssertionError``` is raised when the **input** is **not an integer**, including the exact message ("num must be an integer").

File: solutions/check_odd_or_even.py (hunk 2 above)
Comment (Bikaze) [reply solutions/check_odd_or_even.py]: Great one. Done!

File: solutions/check_odd_or_even.py (hunk 3 above)
Comment (Bikaze) [reply solutions/check_odd_or_even.py:25]: This is the way I know it works;
Raises: (what?)
   AssertionError (ok an assertion error, when?): (Explain when this happens)
I learnt this from a bootcamp. Otherwise, What's your suggestion? Give an example, please.

File: solutions/tests/test_check_odd_or_even.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        self.assertEqual(check_odd_or_even(-10), "Even")
+
+    def test_negative_odd(self):
+        """It should return 'Odd' for a negative odd number."""
+        self.assertEqual(check_odd_or_even(-3), "Odd")
+
+    # Defensive tests
+    def test_non_integer_raises_assertion(self):
+        """It should raise AssertionError when passing a non-integer."""
+        with self.assertRaises(AssertionError):
```
Comment (ObayCipher) [inline solutions/tests/test_check_odd_or_even.py:44]: Defensive test should verify the exception message in``` test_non_integer_raises_assertion``` to rise
```num must be an integer```

File: solutions/tests/test_check_odd_or_even.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    # Standard test cases
+    def test_even_number(self):
+        """It should return 'Even' for an even integer."""
+        self.assertEqual(check_odd_or_even(4), "Even")
+
+    def test_odd_number(self):
+        """It should return 'Odd' for an odd integer."""
+        self.assertEqual(check_odd_or_even(7), "Odd")
+
+    # Edge cases
```
[the remark this answers is not in the document]
Comment (Bikaze) [reply solutions/tests/test_check_odd_or_even.py:28]: Done!

File: solutions/tests/test_check_odd_or_even.py (hunk 4 above)
Comment (Bikaze) [reply solutions/tests/test_check_odd_or_even.py]: Done!

File: solutions/check_odd_or_even.py (hunk 3 above)
Comment (ObayCipher) [reply solutions/check_odd_or_even.py:25]: Just for clear message bout the error that rise from something is not integer it fixed

**Review by ObayCipher — APPROVED** [review]
Great work
